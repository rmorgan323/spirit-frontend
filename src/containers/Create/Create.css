/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.Create {
  width: 400px;
  margin: 20px auto; }
  .Create .clinic-span {
    margin-right: 10px;
    font-family: "Josefin Sans", sans-serif;
    font-size: 14px;
    color: #a7a9ac; }
  .Create ::placeholder {
    font-size: 18px; }
  .Create .new-clinic-form {
    margin-top: 20px; }
  .Create .add-clinic-directions {
    font-family: "Lato", sans-serif;
    font-size: 16px;
    padding: 10px 0; }
  .Create form {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px; }
  .Create .input-clinic-name, .Create .submit-button {
    height: 40px;
    margin: 10px 0;
    border: none; }
  .Create .input-abbreviation::placeholder {
    font-size: 18px;
    letter-spacing: 0px;
    position: relative;
    right: 12px;
    color: #808285; }
  .Create .input-clinic-name, .Create .input-abbreviation {
    padding: 5px 10px;
    color: #6d6e71;
    text-align: center;
    font-size: 20px;
    background-color: #f1f2f2;
    border-bottom: 10px solid #d1d3d4; }
  .Create .input-clinic-name ::placeholder {
    color: #808285; }
  .Create .input-abbreviation {
    height: 80px;
    border: none;
    font-size: 70px;
    padding-left: 37px;
    letter-spacing: 30px;
    border-bottom: 10px solid #d1d3d4; }
  .Create .submit-button {
    background-color: #9E87BA;
    cursor: pointer;
    color: #7F60A4;
    transition: all .2s ease-in-out;
    font-weight: 900;
    color: #fff; }
  .Create .submit-button:hover {
    background-color: #5E378C; }
  .Create .error-message, .Create .success-message {
    font-family: "Josefin Sans", sans-serif;
    font-size: 14px;
    display: flex;
    justify-content: center;
    padding: 10px;
    margin-bottom: 20px; }
  .Create .error-message {
    color: #C00E0E; }
  .Create .success-message {
    color: #5cba47; }
  .Create a.join-link {
    text-decoration: none;
    color: #5cba47; }
  .Create a.join-link:hover {
    color: #468c41; }
