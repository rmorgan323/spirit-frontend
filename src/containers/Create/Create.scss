@import '../../variables.scss';

.Create {
  @include small-content;

  .clinic-span {
    margin-right: 10px;
    @include match-h5;
    color: $grey3;
  }

  ::placeholder {
    font-size: 18px;
  }

  .new-clinic-form {
    margin-top: 20px;
  }

  .add-clinic-directions {
    @include match-p;
    padding: 10px 0;
  }

  form {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;
  }

  .input-clinic-name, .submit-button {
    height: 40px;
    margin: 10px 0;
    border: none;

  }

  .input-abbreviation::placeholder {
    font-size: 18px;
    letter-spacing: 0px;
    position: relative;
    right: 12px;
    color: $grey4;
  }

  .input-clinic-name, .input-abbreviation {
    padding: 5px 10px;
    color: $grey5;
    text-align: center;
    font-size: 20px;
    background-color: $grey1;
    border-bottom: 10px solid $grey2;
  }

  .input-clinic-name {
    ::placeholder {
      color: $grey4;
    }
  }

  .input-abbreviation {
    height: 80px;
    border: none;
    font-size: 70px;
    padding-left: 37px;
    letter-spacing: 30px;
    border-bottom: 10px solid $grey2;
  }

  .submit-button {
    background-color: $purple4;
    cursor: pointer;
    color: $purple5;
    @include std-transition;
    font-weight: 900;
    color: #fff;

  }

  .submit-button:hover {
    background-color: $purple6;
  }

  .error-message, .success-message {
    @include match-h5;
    display: flex;
    justify-content: center;
    padding: 10px;
    margin-bottom: 20px;
  }

  .error-message {
    color: #C00E0E;
  }

  .success-message {
    color: $green4;
  }

  a.join-link {
    text-decoration: none;
    color: $green4;
  }

  a.join-link:hover {
    color: $green5;
  }
} 