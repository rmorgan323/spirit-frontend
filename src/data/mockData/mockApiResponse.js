const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;