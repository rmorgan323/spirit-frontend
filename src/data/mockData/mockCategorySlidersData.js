/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

