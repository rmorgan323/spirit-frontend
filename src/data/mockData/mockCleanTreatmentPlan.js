/*eslint-disable camelcase*/

export const mockCleanTreatmentPlan = {
  id: 10,
  modulation_sensory: null,
  modulation_task: null,
  modulation_environment: null,
  modulation_predictability: null,
  modulation_self_regulation: null,
  modulation_interaction: null,
  modulation_JRC_AR_notes: null,
  posture_sensory: null,
  posture_task: null,
  posture_environment: null,
  posture_predictability: null,
  posture_self_regulation: null,
  posture_interaction: null,
  posture_JRC_AR_notes: null,
  sensory_discrimination_sensory: null,
  sensory_discrimination_task: null,
  sensory_discrimination_environment: null,
  sensory_discrimination_predictability: null,
  sensory_discrimination_self_regulation: null,
  sensory_discrimination_interaction: null,
  sensory_discrimination_JRC_AR_notes: null,
  social_emotional_sensory: null,
  social_emotional_task: null,
  social_emotional_environment: null,
  social_emotional_predictability: null,
  social_emotional_self_regulation: null,
  social_emotional_interaction: null,
  social_emotional_JRC_AR_notes: null,
  executive_functioning_sensory: null,
  executive_functioning_task: null,
  executive_functioning_environment: null,
  executive_functioning_predictability: null,
  executive_functioning_self_regulation: null,
  executive_functioning_interaction: null,
  executive_functioning_JRC_AR_notes: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.468Z",
  updated_at: "2018-02-20T21:25:45.468Z"
};
