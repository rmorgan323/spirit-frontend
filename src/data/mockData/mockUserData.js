/*eslint-disable camelcase*/

export const mockUserData = {
  id: 1,
  authrocket_id: "usr_0vBdVVz3RUsbR2ArDCoPvo",
  name: "Lee Trevino",
  email: "sample@samplemail.com",
  admin: false,
  created_at: "2018-02-19T03:41:23.701Z",
  updated_at: "2018-02-19T03:41:23.701Z",
  clinic: "I Need Therapy",
  clinic_abbreviation: "INT",
  clinic_id: 10,
  clinic_passcode: "utk85fmf"
};
