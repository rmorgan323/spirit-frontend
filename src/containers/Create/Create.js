/*eslint-disable react/no-unescaped-entities*/

import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import InstructionWrapper from '../../components/InstructionWrapper/InstructionWrapper';

import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import * as actions from '../../actions';
import generator from 'generate-password';
import { createClinic } from '../../language/language';
import './Create.css';

export class Create extends Component {
  state = {
    clinicName: '',
    clinicAbbreviation: '',
    successMessage: '',
    error: '',
    success: ''
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  saveNewClinic = (event, name, abbreviation) => {
    event.preventDefault();
    const { saveClinic, user } = this.props;
    const { clinicName, clinicAbbreviation } = this.state;

    let error;
    let success;

    if (!clinicName || !clinicAbbreviation) {
      success = '';
      error = createClinic.clinicNameError;
      this.setState({ error, success });
      return;
    } else if (clinicAbbreviation.length !== 3) {
      success = '';
      error = createClinic.clinicAbbreviationError;
      this.setState({ error, success });
      return;
    }

    const passcode = generator.generate({
      length: 8,
      uppercase: false,
      numbers: true,
      excludeSimilarCharacters: true
    });

    error = '';
    success = createClinic.addClinicSuccess(passcode);
    this.setState({ error, success, clinicName: '', clinicAbbreviation: '' });
    saveClinic({ name, abbreviation, passcode }, user.id);
  };

  displayClinic = () => {
    const { user } = this.props;

    return (
      <div>
        <h3 className="current-clinic-header">{createClinic.currentClinic}</h3>
        <h4>
          <span className="clinic-span">{createClinic.memberOfClinic}</span>
          {user.clinic}
        </h4>
        <h4>
          <span className="clinic-span">{createClinic.clinicAbbreviation}</span>
          {user.clinic_abbreviation}
        </h4>
        <h4>
          <span className="clinic-span">{createClinic.clinicPasscode}</span>
          {user.clinic_passcode}
        </h4>
      </div>
    );
  };

  render() {
    const { clinicName, clinicAbbreviation, error, success } = this.state;
    const { user } = this.props;

    return (
      <div className="Create">
        {user.clinic && this.displayClinic()}

        <h3>Add a New Clinic</h3>

        <InstructionWrapper>
          <div className={`instructions`}>{createClinic.addDirections1}</div>
          <div className={`instructions`}>{createClinic.addDirections2}</div>
        </InstructionWrapper>

        <form className="new-clinic-form">
          <input
            className="input-clinic-name"
            onChange={event => this.handleChange(event)}
            value={clinicName}
            name="clinicName"
            placeholder={createClinic.newClinicPlaceholder}
            maxLength={30}
          />

          <input
            className="input-abbreviation"
            onChange={event => this.handleChange(event)}
            value={clinicAbbreviation}
            name="clinicAbbreviation"
            placeholder={createClinic.clinicAbbreviationPlaceholder}
            maxLength={3}
          />

          <button
            className="submit-button"
            type="submit"
            onClick={event =>
              this.saveNewClinic(event, clinicName, clinicAbbreviation)
            }
          >
            {createClinic.submitClinic}
          </button>
        </form>

        {error && <span className="error-message">{error}</span>}

        {success && <span className="success-message">{success}</span>}

        <NavLink className="join-link" to={`/spirit/users/${user.id}/join`}>
          {createClinic.joinClinic}
        </NavLink>
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  user: store.user
});

export const mapDispatchToProps = dispatch => ({
  saveClinic: (clinic, userId) => {
    dispatch(actions.saveClinic(clinic, userId));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Create);

Create.propTypes = {
  saveClinic: PropTypes.func,
  user: PropTypes.object
};
