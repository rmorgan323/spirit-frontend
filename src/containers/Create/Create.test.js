import React from 'react';
import { Create, mapStateToProps, mapDispatchToProps } from './Create';
import { shallow } from 'enzyme';
import { mockUserData } from '../../data/mockData/mockUserData';

describe('Create tests', () => {
  it('should match snapshot', () => {
    const renderedApp = shallow(<Create user={mockUserData} />);

    expect(renderedApp).toMatchSnapshot();
  });

  it('should set state when clinic name input is changed', () => {
    const renderedApp = shallow(<Create user={mockUserData} />);
    const mockEvent = {
      target: { name: 'clinicName', value: 'I Need Therapy' }
    };

    renderedApp.instance().handleChange(mockEvent);
    expect(renderedApp.state('clinicName')).toEqual('I Need Therapy');
  });

  it('should set state when abbreviation input is changed', () => {
    const renderedApp = shallow(<Create user={mockUserData} />);
    const mockEvent = { target: { name: 'clinicAbbreviation', value: 'INT' } };

    renderedApp.instance().handleChange(mockEvent);
    expect(renderedApp.state('clinicAbbreviation')).toEqual('INT');
  });

  it('should display a success message when saveNewClinic is called with correct params', () => {
    const renderedApp = shallow(
      <Create user={mockUserData} saveClinic={jest.fn()} />
    );
    const mockEvent = { preventDefault: jest.fn() };
    const mockClinicName = 'I Need Therapy';
    const mockClinicAbbreviation = 'INT';

    renderedApp
      .instance()
      .setState({ clinicName: 'I Need Therapy', clinicAbbreviation: 'INT' });
    renderedApp
      .instance()
      .saveNewClinic(mockEvent, mockClinicName, mockClinicAbbreviation);
    expect(renderedApp.state('success').length).toEqual(60);
    expect(renderedApp.state('error').length).toEqual(0);
  });

  it('should display an error message when saveNewClinic is called with incorrect params', () => {
    const renderedApp = shallow(
      <Create user={mockUserData} saveClinic={jest.fn()} />
    );
    const mockEvent = { preventDefault: jest.fn() };
    const mockClinicName = 'I Need Therapy';
    const mockClinicAbbreviation = 'INT';

    renderedApp.instance().setState({ clinicName: '', clinicAbbreviation: '' });
    renderedApp
      .instance()
      .saveNewClinic(mockEvent, mockClinicName, mockClinicAbbreviation);
    expect(renderedApp.state('success').length).toEqual(0);
    expect(renderedApp.state('error').length).toEqual(43);
  });

  it('inputs should be cleared after saving a new clinic', () => {
    const renderedApp = shallow(
      <Create user={mockUserData} saveClinic={jest.fn()} />
    );
    const mockEvent = { preventDefault: jest.fn() };
    const mockClinicName = 'I Need Therapy';
    const mockClinicAbbreviation = 'INT';

    renderedApp
      .instance()
      .setState({ clinicName: 'I Need Therapy', clinicAbbreviation: 'INT' });
    renderedApp
      .instance()
      .saveNewClinic(mockEvent, mockClinicName, mockClinicAbbreviation);
    expect(renderedApp.state('clinicName')).toEqual('');
    expect(renderedApp.state('clinicAbbreviation')).toEqual('');
  });
});

describe('mapStateToProps tests', () => {
  it('should pull user from store', () => {
    const user = { mockUserData };
    const mockStore = { user };
    const result = mapStateToProps(mockStore);

    expect(result.user).toEqual(mockStore.user);
  });
});

describe('mapDispatchToProps tests', () => {
  it('should call dispatch when saveClinic is called', () => {
    const mockDispatch = jest.fn();
    const mockParams = ({
      clinicName: 'I Need Therapy',
      clinicAbbreviation: 'INT',
      passcode: 'fj7f82jb'
    }, 5);
    const result = mapDispatchToProps(mockDispatch);

    result.saveClinic(mockParams);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
