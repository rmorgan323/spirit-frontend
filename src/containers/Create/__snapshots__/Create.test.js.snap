// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Create
    user={
      Object {
        "admin": false,
        "authrocket_id": "usr_0vBdVVz3RUsbR2ArDCoPvo",
        "clinic": "I Need Therapy",
        "clinic_abbreviation": "INT",
        "clinic_id": 10,
        "clinic_passcode": "utk85fmf",
        "created_at": "2018-02-19T03:41:23.701Z",
        "email": "sample@samplemail.com",
        "id": 1,
        "name": "Lee Trevino",
        "updated_at": "2018-02-19T03:41:23.701Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3
            className="current-clinic-header"
          >
            Your Current Clinic
          </h3>
          <h4>
            <span
              className="clinic-span"
            >
              You are a member of: 
            </span>
            I Need Therapy
          </h4>
          <h4>
            <span
              className="clinic-span"
            >
              Clinic Abbreviation: 
            </span>
            INT
          </h4>
          <h4>
            <span
              className="clinic-span"
            >
              Clinic Passcode: 
            </span>
            utk85fmf
          </h4>
        </div>,
        <h3>
          Add a New Clinic
        </h3>,
        <InstructionWrapper>
          <div
            className="instructions"
          >
            To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.
          </div>
          <div
            className="instructions"
          >
            After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.
          </div>
        </InstructionWrapper>,
        <form
          className="new-clinic-form"
        >
          <input
            className="input-clinic-name"
            maxLength={30}
            name="clinicName"
            onChange={[Function]}
            placeholder="New Clinic Name"
            value=""
          />
          <input
            className="input-abbreviation"
            maxLength={3}
            name="clinicAbbreviation"
            onChange={[Function]}
            placeholder="Choose a 3-letter abbreviation for your clinic"
            value=""
          />
          <button
            className="submit-button"
            onClick={[Function]}
            type="submit"
          >
            SUBMIT
          </button>
        </form>,
        "",
        "",
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="join-link"
          to="/spirit/users/1/join"
        >
          JOIN EXISTING CLINIC
        </NavLink>,
      ],
      "className": "Create",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="current-clinic-header"
            >
              Your Current Clinic
            </h3>,
            <h4>
              <span
                className="clinic-span"
              >
                You are a member of: 
              </span>
              I Need Therapy
            </h4>,
            <h4>
              <span
                className="clinic-span"
              >
                Clinic Abbreviation: 
              </span>
              INT
            </h4>,
            <h4>
              <span
                className="clinic-span"
              >
                Clinic Passcode: 
              </span>
              utk85fmf
            </h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Current Clinic",
              "className": "current-clinic-header",
            },
            "ref": null,
            "rendered": "Your Current Clinic",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="clinic-span"
                >
                  You are a member of: 
                </span>,
                "I Need Therapy",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "You are a member of: ",
                  "className": "clinic-span",
                },
                "ref": null,
                "rendered": "You are a member of: ",
                "type": "span",
              },
              "I Need Therapy",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="clinic-span"
                >
                  Clinic Abbreviation: 
                </span>,
                "INT",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Clinic Abbreviation: ",
                  "className": "clinic-span",
                },
                "ref": null,
                "rendered": "Clinic Abbreviation: ",
                "type": "span",
              },
              "INT",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="clinic-span"
                >
                  Clinic Passcode: 
                </span>,
                "utk85fmf",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Clinic Passcode: ",
                  "className": "clinic-span",
                },
                "ref": null,
                "rendered": "Clinic Passcode: ",
                "type": "span",
              },
              "utk85fmf",
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a New Clinic",
        },
        "ref": null,
        "rendered": "Add a New Clinic",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="instructions"
            >
              To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.
            </div>,
            <div
              className="instructions"
            >
              After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.",
              "className": "instructions",
            },
            "ref": null,
            "rendered": "To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.",
              "className": "instructions",
            },
            "ref": null,
            "rendered": "After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.",
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-clinic-name"
              maxLength={30}
              name="clinicName"
              onChange={[Function]}
              placeholder="New Clinic Name"
              value=""
            />,
            <input
              className="input-abbreviation"
              maxLength={3}
              name="clinicAbbreviation"
              onChange={[Function]}
              placeholder="Choose a 3-letter abbreviation for your clinic"
              value=""
            />,
            <button
              className="submit-button"
              onClick={[Function]}
              type="submit"
            >
              SUBMIT
            </button>,
          ],
          "className": "new-clinic-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-clinic-name",
              "maxLength": 30,
              "name": "clinicName",
              "onChange": [Function],
              "placeholder": "New Clinic Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-abbreviation",
              "maxLength": 3,
              "name": "clinicAbbreviation",
              "onChange": [Function],
              "placeholder": "Choose a 3-letter abbreviation for your clinic",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SUBMIT",
              "className": "submit-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "SUBMIT",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "JOIN EXISTING CLINIC",
          "className": "join-link",
          "to": "/spirit/users/1/join",
        },
        "ref": null,
        "rendered": "JOIN EXISTING CLINIC",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3
              className="current-clinic-header"
            >
              Your Current Clinic
            </h3>
            <h4>
              <span
                className="clinic-span"
              >
                You are a member of: 
              </span>
              I Need Therapy
            </h4>
            <h4>
              <span
                className="clinic-span"
              >
                Clinic Abbreviation: 
              </span>
              INT
            </h4>
            <h4>
              <span
                className="clinic-span"
              >
                Clinic Passcode: 
              </span>
              utk85fmf
            </h4>
          </div>,
          <h3>
            Add a New Clinic
          </h3>,
          <InstructionWrapper>
            <div
              className="instructions"
            >
              To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.
            </div>
            <div
              className="instructions"
            >
              After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.
            </div>
          </InstructionWrapper>,
          <form
            className="new-clinic-form"
          >
            <input
              className="input-clinic-name"
              maxLength={30}
              name="clinicName"
              onChange={[Function]}
              placeholder="New Clinic Name"
              value=""
            />
            <input
              className="input-abbreviation"
              maxLength={3}
              name="clinicAbbreviation"
              onChange={[Function]}
              placeholder="Choose a 3-letter abbreviation for your clinic"
              value=""
            />
            <button
              className="submit-button"
              onClick={[Function]}
              type="submit"
            >
              SUBMIT
            </button>
          </form>,
          "",
          "",
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="join-link"
            to="/spirit/users/1/join"
          >
            JOIN EXISTING CLINIC
          </NavLink>,
        ],
        "className": "Create",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="current-clinic-header"
              >
                Your Current Clinic
              </h3>,
              <h4>
                <span
                  className="clinic-span"
                >
                  You are a member of: 
                </span>
                I Need Therapy
              </h4>,
              <h4>
                <span
                  className="clinic-span"
                >
                  Clinic Abbreviation: 
                </span>
                INT
              </h4>,
              <h4>
                <span
                  className="clinic-span"
                >
                  Clinic Passcode: 
                </span>
                utk85fmf
              </h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Current Clinic",
                "className": "current-clinic-header",
              },
              "ref": null,
              "rendered": "Your Current Clinic",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="clinic-span"
                  >
                    You are a member of: 
                  </span>,
                  "I Need Therapy",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "You are a member of: ",
                    "className": "clinic-span",
                  },
                  "ref": null,
                  "rendered": "You are a member of: ",
                  "type": "span",
                },
                "I Need Therapy",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="clinic-span"
                  >
                    Clinic Abbreviation: 
                  </span>,
                  "INT",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Clinic Abbreviation: ",
                    "className": "clinic-span",
                  },
                  "ref": null,
                  "rendered": "Clinic Abbreviation: ",
                  "type": "span",
                },
                "INT",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="clinic-span"
                  >
                    Clinic Passcode: 
                  </span>,
                  "utk85fmf",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Clinic Passcode: ",
                    "className": "clinic-span",
                  },
                  "ref": null,
                  "rendered": "Clinic Passcode: ",
                  "type": "span",
                },
                "utk85fmf",
              ],
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a New Clinic",
          },
          "ref": null,
          "rendered": "Add a New Clinic",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="instructions"
              >
                To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.
              </div>,
              <div
                className="instructions"
              >
                After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.",
                "className": "instructions",
              },
              "ref": null,
              "rendered": "To add a new clinic, you must provide a clinic name and three letter abbreviation. This abbreviation will be used to abstract your patient's real name when adding them to SpIRiT©.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.",
                "className": "instructions",
              },
              "ref": null,
              "rendered": "After submission, you will be provided with a passcode. This passcode will be used by other members in order to join your clinic.",
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-clinic-name"
                maxLength={30}
                name="clinicName"
                onChange={[Function]}
                placeholder="New Clinic Name"
                value=""
              />,
              <input
                className="input-abbreviation"
                maxLength={3}
                name="clinicAbbreviation"
                onChange={[Function]}
                placeholder="Choose a 3-letter abbreviation for your clinic"
                value=""
              />,
              <button
                className="submit-button"
                onClick={[Function]}
                type="submit"
              >
                SUBMIT
              </button>,
            ],
            "className": "new-clinic-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-clinic-name",
                "maxLength": 30,
                "name": "clinicName",
                "onChange": [Function],
                "placeholder": "New Clinic Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-abbreviation",
                "maxLength": 3,
                "name": "clinicAbbreviation",
                "onChange": [Function],
                "placeholder": "Choose a 3-letter abbreviation for your clinic",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SUBMIT",
                "className": "submit-button",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "SUBMIT",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "JOIN EXISTING CLINIC",
            "className": "join-link",
            "to": "/spirit/users/1/join",
          },
          "ref": null,
          "rendered": "JOIN EXISTING CLINIC",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
